dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -53,38 +55,148 @@ public Pattern getParameterPattern() {
+                    return Direction.INPUT;
+                case DatabaseMetaData.functionReturn:
+                    return Direction.RETURN_VALUE;
+            }
+        } else {
+            switch (columnType) {
+                case DatabaseMetaData.procedureColumnIn:
+                    return Direction.INPUT;
+                case DatabaseMetaData.procedureColumnInOut:
+                    return Direction.INPUT_OUTPUT;
+                case DatabaseMetaData.procedureColumnOut:
+                    return Direction.OUTPUT;
+            }
+        }
+        throw new UnsupportedOperationException("Type " + columnType
+                      + " is not a supported type for "
+                      + (isFunction ? "function" : "procedure") + " parameters");
+    }
+
+    private DbParameterAccessor createObjectDbParamAccessor(String paramName,
+                                                            ResultSet rs,
+                                                            boolean isFunction)
+                                                            throws SQLException {
+        Direction paramDirection = getParamDirection(rs.getInt("COLUMN_TYPE"), isFunction);
+        String paramTypeName = rs.getString("TYPE_NAME");
+        int position = rs.getInt("ORDINAL_POSITION"); // We rely upon Derby returning 0 for function return value.
+        return createDbParameterAccessor(paramName, paramDirection,
+                   typeMapper.getJDBCSQLTypeForDBType(paramTypeName), getJavaClass(paramTypeName),
+                   position - 1);
+    }
+
+    public Map<String, DbParameterAccessor> getAllProcedureParameters(String procName)
+                                                                   throws SQLException {
+        // https://github.com/apache/derby/blob/10.13/java/engine/org/apache/derby/impl/jdbc/metadata.properties
+        String sql =
+        "SELECT CAST(NULL AS VARCHAR(128)) AS FUNCTION_CAT," +
+        "       S.SCHEMANAME         AS FUNCTION_SCHEM," +
+        "       A.ALIAS              AS FUNCTION_NAME," +
+        "       V.\"COLUMN_NAME\"    AS COLUMN_NAME," +
+        "       V.\"COLUMN_TYPE\"    AS COLUMN_TYPE," +
+        "       V.\"DATA_TYPE\"      AS DATA_TYPE," +
+        "       V.\"TYPE_NAME\"      AS TYPE_NAME," +
+        "       V.\"PRECISION\"      AS PRECISION," +
+        "       V.\"LENGTH\"         AS LENGTH," +
+        "       CASE WHEN (V.\"DATA_TYPE\" IN (" + java.sql.Types.DECIMAL + "," +
+        "        " + java.sql.Types.NUMERIC + ", " + java.sql.Types.INTEGER + "," +
+        "        " + java.sql.Types.SMALLINT + ", " + java.sql.Types.TINYINT + ", " +
+        "        " + java.sql.Types.BIGINT + ", " + java.sql.Types.DATE + ", " +
+        "        " + java.sql.Types.TIME + ", " + java.sql.Types.TIMESTAMP + "))" +
+        "            THEN" +
+        "            V.\"SCALE\"" +
+        "            ELSE CAST (NULL AS SMALLINT) END" +
+        "            AS SCALE," +
```

Comment:
Many of the columns here are not used. If the query after all is still too hairy, might make some sense to stored in a resource file.